src/lib/components/InfiniteScroll.svelte

```diff
@@ -42,57 +43,36 @@
     options
   );
 
+  // We disconnect previous observer first. We do want to trigger an intersection in case of layout shifting.
+  beforeUpdate(() => observer.disconnect());
+
   afterUpdate(() => {
     // The DOM has been updated. We reset the observer to the current last HTML element of the infinite list.
 
-    // We disconnect previous observer first. We do want to observe multiple elements.
-    observer.disconnect();
-
-    if (container.children.length === 0) {
-      return;
-    }
-
-    const pageIndex: number = container.children.length / pageLimit - 1;
-    // If the pageIndex is not an integer the all page was not fetched - e.g. 50 elements instead of 100 - therefore there is no more elements to fetch
-    if (!Number.isInteger(pageIndex)) {
+    // If not children, no element to observe
+    if (container.lastElementChild === null) {
       return;
     }
 
-    /**
-     * The infinite scroll observe an element that finds place after x % of last page.
-     *
-     * For example given following list of elements:
-     *
-     * [0-100]
-     * [101-200]
-     *
-     * If ratio is set to `0.2`, the observer observes 20% of the last page aka element at position 120.
-     *
-     * [0-100]
-     * [101-200]
-     * [201-300]
-     *
-     * Infinite scroll observe element 220.
-     *
-     * [0-100]
-     * [101-200]
-     * [201-300]
-     * [301-345]
-     *
-     * Infinite scroll does not observe because all data are fetched.
-     */
-    const element: Element | undefined = Array.from(container.children)[
-      pageIndex * pageLimit + Math.round(pageLimit * INFINITE_SCROLL_OFFSET)
-    ];
-
-    if (element === undefined) {
+    // If the infinite scroll is disabled, no observation should happen
+    if (disabled) {
       return;
     }
 
-    observer.observe(element);
+    observer.observe(container.lastElementChild);
   });
 
   onDestroy(() => observer.disconnect());
+
+  // The infinite scroll "disabled" property is updated. If it becomes "false", the observer should be disconnected.
+  $: disabled,
+    () => {
+      if (!disabled) {
```

Comment:
Isn't this covered by `beforeUpdate` callback?

Reply:
You are right I think, removed that auto-subscriber. Thanks!